contracts/MemeFactory.sol

```diff
@@ -341,24 +392,45 @@ abstract contract MemeFactory {
         // Update meme token map values
         totalNativeTokenCommitted += msg.value;
         memeSummon.nativeTokenContributed = totalNativeTokenCommitted;
-        memeHearters[memeToken][msg.sender] += msg.value;
+        memeHearters[memeNonce][msg.sender] += msg.value;
 
         // Record msg.sender activity
         mapAccountActivities[msg.sender]++;
 
-        emit Hearted(msg.sender, memeToken, msg.value);
+        emit Hearted(msg.sender, memeNonce, msg.value);
 
         _locked = 1;
     }
 
+    /// @dev Create a new meme token.
+    function _createThisMeme(
+        string memory name,
+        string memory symbol,
+        uint256 totalSupply
+    ) internal returns (address memeToken) {
+        uint256 localNonce = _nonce;
+        bytes32 randomNonce = bytes32(uint256(keccak256(abi.encodePacked(block.timestamp, msg.sender, localNonce))));
+        bytes memory payload = abi.encodePacked(type(Meme).creationCode, abi.encode(name, symbol, DECIMALS, totalSupply));
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            memeToken := create2(0x0, add(0x20, payload), mload(payload), randomNonce)
+        }
+
+        // Check for non-zero token address
+        require(memeToken != address(0), "Token creation failed");
+
+        // Update nonce
+        _nonce = localNonce + 1;
```

Comment:
Why? In summon we increase. Shouldn't increase it here.

Reply:
We need to increase after each operation. Nonce must be different each time.

Reply:
No you need to increase nonce once per token, not both at summon and unleash

Reply:
ok, but it does not matter at all